src/CommunityToolkit.Maui/Helpers/LogHelpers.shared.cs

```diff
@@ -0,0 +1,28 @@
+using System.Runtime.CompilerServices;
+using CommunityToolkit.Maui.Categories;
+
+namespace CommunityToolkit.Maui.Helpers;
+
+#pragma warning disable IDE0040 // Add accessibility modifiers
+internal static class LogHelpers
```

Comment:
In some situations the user (developer) might want to know what is going on with a specific value he/she puts on the markup extension, that's the main purpose for the log. It is for debugging purposes.

Reply:
In that case we should use Debug.WriteLine, so we do not affect release build.
I would prefer to use standard .net ILogger

Reply:
Ok, up to you. ✌️

But I cannot use standard logger (ILogger) from whithin non DI classes.
The OnScreenSizeExtension class (which inherits from IMarkupExtension) only accepts default constructor, otherwise maui xaml engine throws an error requiring the default constructor.
How to use standard.net logger which depends on DI, on a non DI class?

Of course there is a bunch of workaround such as have access to the container (IserviceProvider) using static methods which were set during container creation/registration, after the `builder.Build()` method gets called, but we cannot do that from `CommunityToolkit.Maui.AppBuilderExtensions` class because the Buld() method is called from outside this class, on the user/developer project side. 

So we cannot get an instance of the IServiceProvider (see below) class on `CommunityToolkit.Maui.AppBuilderExtensions` because on the `builder.Build()` method is not invoked inside MCT's AppBuilderExtensions (and should not be).

``cs
//this does not happens inside MCT, but on the user/developer code instead, after CommunityToolkit.Maui.AppBuilderExtensions is called.
  var serviceProvider = builder.Build().Service;
``

Reply:
Regarding use of `Debug.WriteLine` the purpose of this log I implemented is to allow a developer/user of MCT to debug the Markup extension, if we use Debug.WriteLine, as you know, if will be stripped of the project during release build.

Reply:
About accessing the service, you can solve the services using [this](https://github.com/davidortinau/WeatherTwentyOne/blob/main/src/WeatherTwentyOne/Services/ServiceExtensions.cs) but I prefer to keep with the old (but gold) `Debug.WriteLine`, I'm not sure but I believe it's optimized for release versions

Reply:
But I can remove the whole log if you want.

> Debug.WriteLine

Same response as before, which is this:

> Regarding use of Debug.WriteLine the purpose of this log I implemented is to allow a developer/user of MCT to debug the Markup extension, if we use Debug.WriteLine, as you know, if will be stripped of the project during release build.

Reply:
> you can solve the services using [this](https://github.com/davidortinau/WeatherTwentyOne/blob/main/src/WeatherTwentyOne/Services/ServiceExtensions.cs)

No, see my comment [here](https://github.com/CommunityToolkit/Maui/pull/523#discussion_r932069575)

Reply:
@carolzbnbr but when you say "It is for debugging purposes.", is during the development phase, right? If that's the case you can use `Console.WriteLine` and the dev will see the massage in the output window

Reply:
Not on a computer right now to check, but i used Console.Writeline, isn't it? 🤔

Reply:
Still not sure if we will keep this class.

Reply:
Can't see a good reason why not having it, as right now its implementation is to avoid repeating code on its callers.

Reply:
Well... Since XCT no one asked us to add a Logger or something like that, so we can assume that our `Debug.WriteLine` is enough, and after an internal discussion, we decided to keep this pattern for this PR.

That said, this doesn't mean that we are closed to any suggestions about this subject. That said if you think this project should provide a `Logger` implementation then open a discussion and we talk about an implementation for the entire project and not only for a particular feature.